src/components/movie-view/movie-view.jsx

```diff
@@ -1,28 +1,158 @@
+    getUserDetails(token) {
+        axios.get(`https://t-dogg-movies-api.herokuapp.com/users/${this.props.user}`, {
+            headers: { Authorization: `Bearer ${token}`}
+        }).then(response => {
+            // Use the response to set the user details in the state variables
+            this.setState({
+                userDetails: response.data,
+                favouriteMovies: response.data.FavouriteMovies
+            });
+        }).catch(function(error) {
+            console.log(error);
+        });
+    }
+
+    // Function for adding this movie to a users favourites list. Makes a post request to the server using information passed in through the props
+    addFavourite() {
+        let token = localStorage.getItem('token');
+        // I'm not sure why I need the first {} (before the headers). but without those empty brackets all my requests returned unauthorized
+        axios.post(`https://t-dogg-movies-api.herokuapp.com/users/${this.props.user}/Movies/${this.props.movie._id}`, {}, {
+            headers: { Authorization: `Bearer ${token}` }
+        }).then(response => {
+            // Set the isFavourite state to true, now that this movie has been added to the list of favourites
+            // this.setState({ isFavourite: true });
+            // window.open refreshes the page to make sure this movie is correctly displaying as a favourite
+            window.open(`/movies/${this.props.movie._id}`, '_self');
+        }).catch(function(error) {
+            console.log(error);
+        });
+    }
+
+    // Function for removing this movie from a users favourites list. Makes a delete request to the server using information passed in through the props
+    removeFavourite() {
+        let token = localStorage.getItem('token');
+        axios.delete(`https://t-dogg-movies-api.herokuapp.com/users/${this.props.user}/Movies/${this.props.movie._id}`, {
+            headers: { Authorization: `Bearer ${token}` }
+        }).then(response => {
+            // Set the isFavourite state to false, now that this movie has been removed from the list of favourites
+            // this.setState({ isFavourite: false });
+            // window.open refreshes the page to make sure this movie is correctly displaying as not a favourite
+            window.open(`/movies/${this.props.movie._id}`, '_self');
+        }).catch(function(error) {
+            console.log(error);
+        });
+    }
+
     render() {
         // Get the movie and the onBackClick details from this objects props
         const { movie, onBackClick } = this.props;
+
+        // This section of code sets a flag which will show a add/remove favourites button depending on if the movie can be found in the users favourites
+        let tempArray = this.state.favouriteMovies;
+        let isFavouriteNew = false
+        if (tempArray.includes(this.props.movie._id)) {
+            isFavouriteNew = true;
+        } else {
+            isFavouriteNew = false;
+        };
+
```

Comment:
This logic should be in the render function. You need to write a function that mutates the state. 
You can use lifecyle methods to trigger this function on mount or on update.